src/Microsoft.AspNet.Mvc.Core/ObjectResult.cs

```diff
@@ -122,25 +116,12 @@ public override Task ExecuteResultAsync(ActionContext context)
                 //
 
                 // 1. Select based on sorted accept headers.
-                if (sortedAcceptHeaderMediaTypes.Any())
-                {
-                    selectedFormatter = SelectFormatterUsingSortedAcceptHeaders(
-                                                                            formatterContext,
-                                                                            formatters,
-                                                                            sortedAcceptHeaderMediaTypes);
-                }
+                selectedFormatter = SelectFormatterUsingSortedAcceptHeaders(
+                    formatterContext,
+                    formatters,
+                    sortedAcceptHeaderMediaTypes);
 
-                // 2. No formatter was found based on accept headers, fall back on request Content-Type header.
-                if (selectedFormatter == null && requestContentType != null)
-                {
-                    selectedFormatter = SelectFormatterUsingAnyAcceptableContentType(
-                                                                                formatterContext,
-                                                                                formatters,
-                                                                                new[] { requestContentType });
-                }
-
-                // 3. No formatter was found based on Accept and request Content-Type headers, so
-                // fallback on type based match.
+                // 2. No formatter was found based on Accept header. Fallback to type-based match.
```

Comment:
:ok: main thing is the formatters are passed `null` for their `MediaTypeHeaderValue` parameter. They can't make any decisions based on _that_ parameter and are left with just the `Type` parameter. Agree they can choose to ignore that.